app/src/main/java/com/android/sample/ui/createRecipe/AddInstructionStepScreen.kt

```diff
@@ -0,0 +1,195 @@
+import com.android.sample.ui.utils.PlateSwipeScaffold
+
+@Composable
+fun AddInstructionStepScreen(
+    navigationActions: NavigationActions,
+    createRecipeViewModel: CreateRecipeViewModel
+) {
+  PlateSwipeScaffold(
+      navigationActions = navigationActions,
+      selectedItem = Route.CREATE_RECIPE,
+      showBackArrow = true,
+      content = { paddingValues ->
+        AddInstructionStepContent(
+            paddingValues = paddingValues,
+            createRecipeViewModel = createRecipeViewModel,
+            navigationActions = navigationActions)
+      })
+}
+
+/**
+ * Composable function that displays the content for adding an instruction step.
+ *
+ * @param paddingValues Padding values to be applied to the content.
+ * @param createRecipeViewModel ViewModel for managing the recipe creation process.
+ * @param navigationActions Actions for navigating between screens.
+ * @param modifier Modifier to be applied to the content.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddInstructionStepContent(
+    paddingValues: PaddingValues,
+    createRecipeViewModel: CreateRecipeViewModel,
+    navigationActions: NavigationActions,
+    modifier: Modifier = Modifier
+) {
+  var stepDescription by remember { mutableStateOf("") }
+  var stepTime by remember { mutableStateOf("") }
+  var stepCategory by remember { mutableStateOf("") }
+  var selectedIcon by remember { mutableStateOf<IconType?>(null) }
+  var showError by remember { mutableStateOf(false) }
+
+  Column(
+      modifier =
+          modifier.fillMaxSize().padding(paddingValues).padding(horizontal = HORIZONTAL_PADDING.dp),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Top) {
+        // Step Label
+        Text(
+            text = stringResource(R.string.step_label),
+            style = Typography.titleLarge,
+            color = MaterialTheme.colorScheme.onPrimary,
+            modifier = Modifier.padding(vertical = 8.dp).testTag("StepLabel"))
+
+        // Container for input fields
+        Box(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .background(color = lightCream, shape = RoundedCornerShape(8.dp))
+                    .padding(16.dp)
+                    .testTag("InputContainer")) {
+              Column {
+                // Row for input fields for time, category, and icon
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.spacedBy(8.dp),
+                    verticalAlignment = Alignment.CenterVertically) {
+                      // Time input field
+                      OutlinedTextField(
+                          value = stepTime,
+                          onValueChange = { stepTime = it },
+                          label = {
+                            Text(stringResource(R.string.time_label), style = Typography.bodySmall)
```

Comment:
It was a good idea to limit the keyboard to numbers only for the time, but there also isn't any indication of what unit this number is in. Was this something to be implemented later?

Reply:
Hey! Thanks for the feedback! 😊 Right now, this is just a raw value, and the handling/limitations for this value will be implemented in a dedicated PR focused on the screen that lists the instructions.

Let me know if you have any other questions!

Reply:
Done in #191